awswrangler/data_quality/_utils.py

```diff
@@ -83,32 +108,38 @@ def _start_ruleset_evaluation_run(
     return cast(str, response["RunId"])
 
 
-def _get_ruleset_evaluation_run(
+def _get_ruleset_run(
     run_id: str,
+    run_type: str,
     boto3_session: Optional[boto3.Session] = None,
 ) -> Dict[str, Any]:
-    boto3_session = _utils.ensure_session(session=boto3_session)
-    client_glue: boto3.client = _utils.client(service_name="glue", session=boto3_session)
-    return cast(Dict[str, Any], client_glue.get_data_quality_ruleset_evaluation_run(RunId=run_id))
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_glue: boto3.client = _utils.client(service_name="glue", session=session)
+    if run_type == "recommendation":
+        response = client_glue.get_data_quality_rule_recommendation_run(RunId=run_id)
+    elif run_type == "evaluation":
+        response = client_glue.get_data_quality_ruleset_evaluation_run(RunId=run_id)
+    return cast(Dict[str, Any], response)
 
 
-def _wait_ruleset_evaluation_run(
+def _wait_ruleset_run(
```

Comment:
Refactored wait to handle both recommendation and evaluation runs since they have the same logic